syntax/coq.vim

```diff
@@ -39,11 +39,10 @@ if !exists('b:coqtail_did_highlight') || !b:coqtail_did_highlight
       hi def CoqtailChecked ctermbg=4 guibg=LightGreen
       hi def CoqtailSent ctermbg=7 guibg=LimeGreen
     endif
-    " TODO: check these with someone...
-    hi def CoqtailDiffAddedBg ctermbg=0 guibg=Black
-    hi def CoqtailDiffRemovedBg ctermbg=0 guibg=Black
-    hi def CoqtailDiffAdded ctermbg=2 guibg=Green
-    hi def CoqtailDiffRemoved ctermbg=1 guibg=Red
+    hi def link CoqtailDiffAdded DiffText
+    hi def link CoqtailDiffRemoved DiffDelete
+    hi def link CoqtailDiffAddedBg DiffChange
+    hi def link CoqtailDiffRemovedBg DiffChange
```

Comment:
Do we want to give users the option to specify their own proof diff highlighting, e.g., using `g:CoqtailHighlight`?

Reply:
If I understand `:h :hi-default` correctly users should be able to override these default settings by putting something like `hi CoqtailDiffAdded ctermfg=1` in their .vimrc or anything that gets sourced before Coqtail's syntax file. For that matter, the same should be true of `CoqtailChecked` and `CoqtailSent` so `g:CoqtailHighlight` shouldn't be necessary at all.

Reply:
I see, so I'll change mine here to use that instead.

Reply:
I tried removing the `g:CoqtailHighlight` from my `.vimrc` (which I used for custom `CoqtailChecked` and `CoqtailSent` highlighting) and it stopped working.. I'm not totally sure why. I'll need to investigate that further but I think we'll want to keep `g:CoqtailHightligh` here for now, but also keep this PR as it is like this?

Reply:
Hm, if I put `hi link CoqtailChecked Error` in my .vimrc (and no `g:CoqtailHighlight`) it shows up as red as expected. In any case, I agree we should leave `g:CoqtailHighlight` for now and go through some sort of deprecation process in a later PR so people aren't surprised by their custom highlighting suddenly not working.